hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java

```diff
@@ -105,14 +124,58 @@ public void validateGossipEndpoint(
      */
     public void validateServiceEndpoint(
             @Nullable final List<ServiceEndpoint> endpointList, @NonNull final NodesConfig nodesConfig) {
+        requireNonNull(nodesConfig);
+
         validateFalse(endpointList == null || endpointList.isEmpty(), INVALID_SERVICE_ENDPOINT);
         validateFalse(endpointList.size() > nodesConfig.maxServiceEndpoint(), INVALID_SERVICE_ENDPOINT);
         for (final var endpoint : endpointList) {
             validateEndpoint(endpoint, nodesConfig);
         }
     }
 
+    /**
+     * Validates the admin key.
+     *
+     * @param key The key to validate
+     * @throws PreCheckException if the key is invalid
+     */
+    public void validateAdminKey(@Nullable Key key) throws PreCheckException {
+        final var keyEmpty = isEmpty(key);
+        validateFalsePreCheck(key == null || keyEmpty, KEY_REQUIRED);
+        validateTruePreCheck(isValid(key), INVALID_ADMIN_KEY);
+    }
+
+    /**
+     * Validates the admin key in the handle.
+     * @param handleContext
+     * @param key
+     */
+    public void validateAdminKeyInHandle(@NonNull final HandleContext handleContext, @NonNull final Key key) {
+        requireNonNull(handleContext);
+        requireNonNull(key);
+
+        try {
+            handleContext.attributeValidator().validateKey(key);
+        } catch (HandleException e) {
+            throw new HandleException(INVALID_ADMIN_KEY);
+        }
+    }
```

Comment:
We should add validateKey that accepts response code in AttributeValidator

Reply:
Added com.hedera.node.app.spi.validation.AttributeValidator#validateKey(com.hedera.hapi.node.base.Key, com.hedera.hapi.node.base.ResponseCodeEnum), removed this method.